jupyterhub_chameleon/utils.py

```diff
@@ -14,40 +10,38 @@ class Artifact:
     """A shared experiment/research artifact that can be spawned in JupyterHub.
 
     Attrs:
-        deposition_repo (str): the name of the deposition repository (e.g.,
-            "zenodo" or "chameleon")
-        deposition_id (str): the ID of the deposition within the repository.
-        id (str): the external Trovi ID of the artifact linked to this
-            deposition. Default = None.
+        contents_id (str): the ID of the artifact's Trovi contents.
+        contents_repo (str): the name of the artifact's Trovi contents repository.
+        contents_url (str): the access URL for the artifact's Trovi contents.
+        contents_proto (str): the protocol for accessing the artifact's Trovi contents.
         ownership (str): the requesting user's ownership status of this
             artifact. Default = "fork".
         ephemeral (bool): whether the artifact's working environment should
             be considered temporary. This can indicate that, e.g., nothing
             from the working directory should be saved when the spawned
             environment is torn down. Default = False.
     """
-    def __init__(self, deposition_repo=None, deposition_id=None, id=None,
-                 contents_url=None, ownership='fork', ephemeral=None,
-                 contents_proto=None):
-        self.id = id
-        self.deposition_repo = deposition_repo
-        self.deposition_id = deposition_id
+
+    def __init__(
+        self,
+        contents_id=None,
+        contents_repo=None,
+        contents_url=None,
+        contents_proto=None,
+        ownership="fork",
+        ephemeral=None,
+    ):
+        self.contents_id = contents_id
+        self.contents_repo = contents_repo
         self.contents_url = contents_url
         self.contents_proto = contents_proto
         self.ownership = ownership
-        self.ephemeral = ephemeral in [True, 'True', 'true', 'yes', '1']
-
-        # Only the deposition information is required. Theoretically this can
-        # allow importing arbitrary Zenodo DOIs or from other sources that are
-        # not yet existing on Trovi.
-        if not (deposition_repo and deposition_id):
-            raise ValueError('Missing deposition information')
+        self.ephemeral = ephemeral in [True, "True", "true", "yes", "1"]
 
-    def deposition_url(self):
-        url_factory = getattr(self.__class__, f'{self.deposition_repo}_url_factory', None)
-        if not url_factory:
-            return None
-        return url_factory(self.deposition_id)
+        # Only the contents information is required. Theoretically this can
+        # allow importing from other sources that are not yet existing on Trovi.
```

Comment:
Can you clarify what you mean? As in you could pass in an archive URL to download?

Also, should the proto be required?

Reply:
Because we pass in the contents URL directly, the "import" handler could actually be used outside of a Trovi context. Like, you could do:

GET /import?contents_url=<tarball>&contents_repo=external

There's a case to be made that the repo is also not really required. To your point, the protocol probably should be b/c it's used to determine how the contents are used.